src/kaocha/plugin/junit_xml.clj

```diff
@@ -100,32 +100,35 @@
                         (hierarchy/fail-type? m) (failure->xml m)))
                     events)}))
 
-(defn suite->xml [suite index]
+(defn suite->xml [{:keys [no-system-out?]} suite index]
   (let [id (::testable/id suite)]
-    {:tag :testsuite
-     :attrs (-> {:name (test-name suite)
-                 :id index
-                 :hostname "localhost"}
-                (merge (stats suite) (time-stat suite))
-                (assoc :package (if (qualified-ident? id)
-                                  (namespace id)
-                                  "")))
+    {:tag     :testsuite
+     :attrs   (-> {:name     (test-name suite)
+                   :id       index
+                   :hostname "localhost"}
+                  (merge (stats suite) (time-stat suite))
+                  (assoc :package (if (qualified-ident? id)
+                                    (namespace id)
+                                    "")))
      :content (concat
-               [{:tag :properties}]
-               (map testcase->xml (leaf-tests suite))
-               [{:tag :system-out
-                 :content (->> suite
-                               test-seq
-                               (keep :kaocha.plugin.capture-output/output)
-                               (remove #{""})
-                               (map strip-ansi-sequences))}
-                {:tag :system-err}])}))
+                [{:tag :properties}]
+                (map testcase->xml (leaf-tests suite))
+                [(merge {:tag :system-out}
+                        (when-not no-system-out?
+                          {:content (->> suite
+                                         test-seq
+                                         (keep :kaocha.plugin.capture-output/output)
+                                         (remove #{""})
+                                         (map strip-ansi-sequences))}))
+                 {:tag :system-err}])}))
 
 (defn result->xml [result]
   (let [suites (::result/tests result)]
     {:tag     :testsuites
      :attrs   {}
-     :content (map suite->xml suites (range))}))
+     :content (map (partial suite->xml {:no-system-out? (::no-system-out? result)})
```

Comment:
Why not stick with the namespaced keyword everywhere? You can destructure it above with `::keys`.